nautobot/docs/development/best-practices.md

```diff
@@ -210,73 +250,76 @@ class UserFilter(NautobotFilterSet):
+    - Fields that use `name` or some other natural key field instead of `slug` can set the `to_field_name` argument on `NaturalKeyOrPKMultipleChoiceFilter` accordingly.
 
 ```python
-    from nautobot.core.filters import NaturalKeyOrPKMultipleChoiceFilter
+# Typical usage
+from nautobot.core.filters import NaturalKeyOrPKMultipleChoiceFilter
+
     provider = NaturalKeyOrPKMultipleChoiceFilter(
         queryset=Provider.objects.all(),
         label="Provider (slug or ID)",
     )
-    # optionally use the to_field_name argument to set the field to name instead of slug
+```
+
+```python
+# Optionally, using the to_field_name argument to look up by "name" instead of by "slug"
+from nautobot.core.filters import NaturalKeyOrPKMultipleChoiceFilter
+
     provider = NaturalKeyOrPKMultipleChoiceFilter(
         to_field_name="name",
         queryset=Provider.objects.all(),
         label="Provider (name or ID)",
     )
 ```
 
-### Filter Naming and Definition
+- Boolean filters for membership **must** be named with `has_{related_name}` (e.g. `has_interfaces`) and should use the `RelatedMembershipBooleanFilter` filter class.
+    - One exception to this naming convention may be made for Boolean filters for identity, which **may** be named `is_{name}` instead (e.g. `is_virtual_chassis_member` versus `has_virtual_chassis`). Although this is semantically identical to `has_` filters, there may be occasions where naming the filter `is_` would be more intuitive.
 
-- Boolean filters for membership **must** be named with `has_{related_name}` (e.g. `has_interfaces`)
+```python
+from nautobot.core.filters import RelatedMembershipBooleanFilter
 
-- Boolean filters for identity **must** be named with `is_{name}` (e.g. `is_virtual_chassis`) although this is semantically identical to `has_` filters, there may be occasions where naming the filter `is_` would be more intuitive.
+    has_interfaces = RelatedMembershipBooleanFilter(
+        field_name="interfaces",
+        label="Has interfaces",
+    )
 
-- Filters **must** declare [`field_name`](https://django-filter.readthedocs.io/en/stable/ref/filters.html#field-name) when they have a different name than the underlying model field they are referencing. Where possible the suffix component of the filter name **must** map directly to the underlying field name.
+    is_virtual_chassis_member = RelatedMembershipBooleanFilter(
+        field_name="virtual_chassis",
+        label="Is a virtual chassis member",
+    )
+```
 
-  For example, `DeviceFilterSet.has_console_ports` could be better named, to assert that the filter name following the `has_` prefix is a one-to-one mapping to the underlying model's related field name (`consoleports`) therefore `field_name` must point to the field name as defined on the model:
+- Whenever possible otherwise, filter names **must** correspond exactly to the underlying model field they are referencing.
```

Comment:
Perhaps handle like Boolean filters and declare the must here and the exception below?

Reply:
Not sure I understand the suggestion. Can you provide more details?